pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -471,6 +508,8 @@ func (m *kubeGenericRuntimeManager) computePodContainerChanges(pod *v1.Pod, podS
 			if pod.Spec.RestartPolicy != v1.RestartPolicyNever {
 				message := fmt.Sprintf("Failed to initialize pod. %q will be restarted.", container.Name)
 				glog.V(1).Info(message)
+
+				m.recorder.Eventf(containerRef, v1.EventTypeWarning, events.ContainerRestarting, message)
```

Comment:
it's possible i missed other areas, but I *think* this is prone to the same problem I fixed in https://github.com/kubernetes/kubernetes/pull/46246/files#diff-44f60dd6d99cb695e5f333647ebd0703R449

I worry we should be adapting these ;-)  i would feel better if we have one common function like generateContainerEvent that is called when good or bad so I know its right everywhere.